docs/code-review/2026-05-03-comprehensive.md

```
@@ -0,0 +1,363 @@
+`(offset_x * offset_x + offset_y * offset_y).sqrt()` is the textbook hand-rolled hypotenuse, and clippy's `imprecise_flops` flags it because `f64::hypot` is more numerically stable (no intermediate overflow) and self-documenting. The values here are tiny pixel offsets so accuracy isn't a real concern, but it's a one-character readability win.
+
+**Proposed fix:**
+`let distance = offset_x.hypot(offset_y);`
+
+#### [rust-idioms] Replace `map(...).unwrap_or(false)` with `is_some_and`
+
+**Severity:** nit
+**Files:** `src/ui/dock_menu.rs:60-63`
+
+**Why this matters:**
+```rust
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)